docs/r1cs-docs-example.md

```diff
@@ -0,0 +1,541 @@
+The rank-1 constraint system API for programmatically defining constraint systems.
+
+## Building a proof-of-shuffle constraint system
+
+A shuffle is a permutation of a list of `k` scalars `{x_i}` into a list of `k` scalars `{y_i}`.
+
+Algebraically it can be expressed as a statement that for a free variable `z`, the roots of the two polynomials in terms of `z` are the same up to a permutation:
```

Comment:
```suggestion
Algebraically it can be expressed as a statement that for a free variable \\(z\\), the roots of the two polynomials in terms of \\(z\\) are the same up to a permutation:
```
katex